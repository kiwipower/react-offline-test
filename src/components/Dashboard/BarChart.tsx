import React from "react";

interface Props {
  generationmix: Array<{ fuel: string; perc: string }>;
}

function BarChart({ generationmix }: Props) {
  return <div>{JSON.stringify(generationmix)}</div>;
}

export default BarChart;
