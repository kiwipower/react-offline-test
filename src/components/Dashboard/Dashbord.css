.container {
    display: flex;
    flex-direction: column;
    max-width: 64rem;
    height: 37.5rem;
    margin: 0 auto;
    flex-grow: 1;
    justify-content: center;
}

.chart-header {
    text-align: center;
    font-size: .8rem;
    padding: 1rem 0;

    @media(min-width: 48rem) {
        font-size: 2rem;
    }
}

.center-flex {
    align-self: center;
}