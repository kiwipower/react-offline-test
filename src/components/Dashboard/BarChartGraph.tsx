import * as React from "react";
import "./Dashbord.css";
import { COLORS } from "../contants";
import {
  BarChart,
  Bar,
  Cell,
  ResponsiveContainer,
  XAxis,
  YAxis,
  Label,
  CartesianGrid,
  Tooltip
} from "recharts";
import { DateViewer } from "../shared";

interface Props {
  energy: {
    generationmix: Array<{ fuel: string; perc: number }>;
    from: string;
    to: string;
  };
}

function BarChartGraph({ energy }: Props) {
  if (!energy) {
    return null;
  }
  const { generationmix, from, to } = energy;

  return (
    <React.Fragment>
      <DateViewer from={from} to={to} className="chart-header" />
      <ResponsiveContainer>
        <BarChart data={generationmix}>
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis dataKey="fuel">
            <Label value="Uk Energy Name" offset={0} position="insideBottom" />
          </XAxis>
          <YAxis dataKey="perc">
            <Label
              value="Uk Energy in percentage (%)"
              angle={-90}
              offset={0}
              position="left"
              style={{ textAnchor: "middle" }}
            />
          </YAxis>
          <Tooltip
            labelFormatter={label => `fuel: ${label}`}
            formatter={value => `${value}%`}
          />
          <Bar dataKey="perc">
            {generationmix.map((entry, index) => (
              <Cell
                key={`${entry.perc}-cell-${index}`}
                fill={COLORS[index]}
                strokeWidth={index === 2 ? 4 : 1}
              />
            ))}
          </Bar>
        </BarChart>
      </ResponsiveContainer>
    </React.Fragment>
  );
}

export default BarChartGraph;
