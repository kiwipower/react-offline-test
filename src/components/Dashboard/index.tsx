import * as React from "react";
import { ErrorMessage, LoadingSpinner } from "../shared";
import Charts from "./BarChartGraph";
import energyApi from "../../api/energyApi";
import "./Dashbord.css";

interface Energy {
  from: string;
  to: string;
  generationmix: Array<{ fuel: string; perc: number }>;
}

interface ErrorMessage {
  message: string;
}

function Dashboard() {
  const [energy, setEnergy] = React.useState<Energy | undefined>();
  const [error, setError] = React.useState<string | undefined>();

  const handleFetchData = () => {
    return energyApi.list().then(
      (response: Energy) => {
        setEnergy(response);
      },
      (error: ErrorMessage) => {
        setError(error.message);
      }
    );
  };

  React.useEffect(() => {
    handleFetchData();
  }, []);

  const isLoading = energy === undefined && error === undefined;

  return (
    <div className="container">
      {isLoading ? (
        <LoadingSpinner className="center-flex" isLoading={isLoading} />
      ) : !isLoading && error ? (
        <ErrorMessage className="center-flex" error={error} />
      ) : (
        <Charts energy={energy} />
      )}
    </div>
  );
}

export default Dashboard;
