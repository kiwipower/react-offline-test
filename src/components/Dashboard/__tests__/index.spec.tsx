import * as React from "react";
import { shallow } from "enzyme";
import Dashboard from "../index";

describe("<Dashboard />", () => {
  const component = shallow(<Dashboard />);

  it("should display loading", () => {
    expect(component.type()).not.toEqual(null);
  });

  it("should render", () => {
    expect(component).toMatchSnapshot();
  });
});
