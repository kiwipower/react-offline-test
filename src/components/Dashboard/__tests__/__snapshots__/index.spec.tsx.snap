// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should render 1`] = `ShallowWrapper {}`;
